src/accelerate/accelerator.py

```diff
@@ -3833,3 +3835,28 @@ def fp8_backend(self):
         elif self.state.deepspeed_plugin is not None and self.state.deepspeed_plugin.enable_msamp:
             return "MSAMP"
         return None
+
+    @staticmethod
+    def compile_regions(module: torch.nn.Module, **compile_kwargs) -> torch.nn.Module:
+        """
+        Compiles regions of the model and returns a new instance of the model without copying parameters (like
+        `torch.compile`).
+        """
+
+        if isinstance(module, torch.nn.ModuleList):
+            # Create a new ModuleList
+            new_modulelist = torch.nn.ModuleList()
+            for submodule in module:
+                new_modulelist.append(torch.compile(submodule, **compile_kwargs))
+            return new_modulelist
+        elif module._modules:  # Non-leaf node
+            # Create a new module of the same type
+            new_module = module.__class__.__new__(module.__class__)
+            new_module.__dict__.update(module.__dict__)
+            new_module._modules = {}
+            # Process children
+            for name, submodule in module.named_children():
+                new_module.add_module(name, TorchDynamoPlugin.compile_regions(submodule, **compile_kwargs))
```

Comment:
I think you mean Accelerator no ? 
```suggestion
                new_module.add_module(name, Accelerator.compile_regions(submodule, **compile_kwargs))
```

Reply:
Yes, do you think this should be in the accelerator or `TorchDynamoPlugin` ?

Reply:
okay, I'll move it there (makes sense to have it close to `is_compiled_module`)